integrated_channels/blackboard/exporters/learner_data.py

```diff
@@ -33,11 +33,15 @@ def get_learner_data_records(
         """
         enterprise_customer_user = enterprise_enrollment.enterprise_customer_user
         if enterprise_customer_user.user_email is None:
-            LOGGER.debug(
-                'No learner data was sent for LMS User [%s] because Blackboard User ID not found for customer [%s]',
+            LOGGER.debug(generate_formatted_log(
+                'blackboard',
+                enterprise_customer_user.enterprise_customer.uuid,
```

Comment:
so when I asked for uuid I meant for the user, not the enterprise.
The key thing I am trying to solve for is getting a ticket from ECS that says "some user (probably their email address) is having some issue" and being able to track them down with *some* identifier. The reason I said ID was pointless was because if I try a search in Splunk that is like `error 6` I will get a lot, A LOT of logs that have nothing to do with a *user* id of 6. Hell I'll get logs that don't even have to do with any ID and just happen to have a 6 in them.
Thus this is why I was trying to insinuate using uuid *instead* of id, which is functionally useless, but was probably not clear enough about that.

So this line would be `enterprise_customer_user.uuid`

Reply:
I think given our high user ids for LMS at this stage, this is probably not actually an issue. And I forgot to mention I do see value in having the enterprise's actual uuid. I'm not 100% sure if we use LMS user id everywhere (we seem to use it almost everywhere), so I'm fine with standardizing on that. In general I'd prefer us using uuids but the direct link from enterprise customer user to LMS is the LMS user id, and I don't know that we'll ever get a ticket about a user that actually has an id less than 100,000

Reply:
EnterpriseCustomerUser does not currently have a UUID field.  The table is only a link between an EnterpriseCustomer (which has a UUID) and the LMS User id, for reasons in the OEP I linked before. I think that it is reasonable to bring up adding a UUID specifically to EnterpriseCustomerUser, maybe as a discovery or Enterprise Arch ticket, but it's not part of this logging ticket. 

Since we don't have a UUID for a user when searching logs, and you might have to filter out a lot of unrelated errors, the other solution I propose that is in scope is I could change the text in the formatter to be less generic to make false positives unlikely:  

Right now the formatter logs `[User: {lms_user_id}]`, so a typical debug process would have to be (1) lookup the LMS user id in the Read Replica by email, then (2) in Splunk search for the text `"Integrated Channel" [User: <value>]`. 

I could change the `[User: <value>]` to be a single, fuller identifier like `[integrated_channel_lms_user: <value>]` for fewer other potential log hits?

Reply:
Ack. You're right - what the heck table was I looking at earlier. I swore I saw a uuid on ECU. Sorry about that!

I think my initial concern was overlap in ids of varying kinds and being able to track a single metric/identifier across different services. This usually involves some unique token attributed to an entire process.
I mostly just want edX to be consistent in that as much as we can be (I know for instance, with pending enterprise users they may or may not have a user id and we *have* to use email addresses).

To your suggestion, that definitely allows for the ability to structure a search to isolate integrated channel logs by user, however, it still requires knowing that entire log prefix ahead of time.
My ideal is being able to take a user id/uuid, drop it into Splunk and get ONLY logs related to that user's interaction and processes in all of edX. If I have a user id, but not entirely sure what I'm looking for is integrated channels related, I will have to sift through results to get an example of that log output (or find this code).

Most likely, the people doing the digging are very familiar with integrated channels and probably already use a search term of `integrated_channels` (I know I sure do!) and our LMS user ids are unique enough at this stage that I can (and have) just plopped one into a Splunk search with no other search terms and began isolating down what I'm looking for, so this is all not actually a big deal at all and I feel non-blocking at this stage. It [standardized logging across systems] is also an initiative that needs to be driven at a top down level (likely by Arch)

Reply:
After writing a different comment, I think your suggestion is a good one just in the fact we can then facet that specific "field" and be able to do something like `integrated_channel_lms_user=<id> "specific error we're hunting down"` in Splunk. We could also then filter just by the user id and see any other integrated channel logs that may or may not be occurring with that user.